Tests/Sources/VLCMediaListTest.swift

```diff
@@ -0,0 +1,55 @@
+/*****************************************************************************
+ * VLCMediaListTest.swift
+ *****************************************************************************
+ * Copyright (C) 2018 Mike JS. Choi
+ * Copyright (C) 2018 VLC authors and VideoLAN
+ * $Id$
+ *
+ * Authors: Mike JS. Choi <mkchoi212 # icloud.com>
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston MA 02110-1301, USA.
+ *****************************************************************************/
+
+import XCTest
+
+class VLCMediaListTest: XCTestCase {
+
+    func testRemoveMedia() throws {
+        let videos = [Video.test1, Video.test1, Video.test2, Video.test3, Video.test4, Video.test4]
+        let source = videos.map{ $0.media }
+        let mediaList = try XCTAssertNotNilAndUnwrap(VLCMediaList(array: source))
+        
+        let NF = UInt(NSNotFound)
+        let tests: [(targetIdx: UInt, expectedState: [UInt], ok: Bool)] = [
+            (0, [NF, 0, 1, 2, 3, 4], true),
+            (3, [NF, 0, 1, 2,NF, 3], true),
+            (9, [NF, 0, 1, 2,NF, 3], false),
+            (2, [NF, 0, 1,NF,NF, 2], true),
+            (1, [NF, 0,NF,NF,NF, 1], true),
+            (8, [NF, 0,NF,NF,NF, 1], false),
+            (0, [NF,NF,NF,NF,NF, 0], true),
+            (0, [NF,NF,NF,NF,NF,NF], true),
```

Comment:
> quick question when you try again to remove at index 0 after removing it already at index 0 shouldn't the result 
be false?

You are completely right :D

> Do we need the NF in expectedState? 

I see where you are coming from and yeah it's a bit hard to match the indices. But I think it would be better to be explicit about exactly which media was removed.

e.g. Let say you have something like this

```swift
let videos = [Video.test1, Video.test1]
let removalResult = mediaList.removeMedia(at: 0)
let currentState = source.map{ mediaList.index(of: $0) }
XCTAssertEqual(currentState, [0])
```

From `XCTAssertEqual(currentState, [0])`, you can't tell which of the two `Video.test1` was removed.